src/app/accusoft-editor.component.js

```diff
@@ -1,22 +1,21 @@
-import React, { useRef, useEffect } from 'react'
+import React, { useRef, useEffect, useState } from 'react'
 
 export default function Editor(props) {
   const container = useRef()
+  const [hasRun, setHasRun] = useState()
   const editor = new PrizmDocEditor({
     baseUrl: 'https://prizmdoc-integ.canopy.ninja/',
     sessionId: props.sessionId
   });
 
   useEffect(() => {
     if (container.current) {
-      (async function () {
-        const editorApi = await editor.embed({
-          container: container.current
-        });
-        props.setEditor(editorApi);
-      })();
+      setHasRun(true)
+      editor.embed({
+        container: container.current
+      }).then(editor => props.setEditor(editor))
     }
-  }, [container.current])
+  }, [container.current, hasRun])
```

Comment:
for some reason, it seems the change to the container.current is causing this to be run twice and thus two insertions?

Reply:
so two iframes I mean